install_script.sh.template

```diff
@@ -330,25 +330,33 @@ if [ -n "$DD_FIPS_MODE" ]; then
 fi
 
 declare -a apm_libraries
+declare host_injection_enabled
+declare docker_injection_enabled
 
-host_injection_enabled=
-if [[ "$DD_APM_HOST_INJECTION_ENABLED" == "true" || ( -z "$DD_APM_HOST_INJECTION_ENABLED" && "$DD_APM_INSTRUMENTATION_ENABLED" == "true") ]] ; then
-  if [ "$no_agent" ]; then
-      ERROR_MESSAGE="Host injection is not supported with the DD_NO_AGENT_INSTALL flag. Disable host injection with DD_APM_HOST_INJECTION_ENABLED=false or unset DD_NO_AGENT_INSTALL"
-      ERROR_CODE=$INVALID_PARAMETERS_CODE
-      echo -e "\033[31m$ERROR_MESSAGE\033[0m\n"
-      report_telemetry
-      exit 1;
-  fi
+if [ "$DD_APM_INSTRUMENTATION_ENABLED" = "all" ] ; then
+  host_injection_enabled="true"
+  docker_injection_enabled="true"
+fi
 
+if [ "$DD_APM_INSTRUMENTATION_ENABLED" = "host" ] ; then
   host_injection_enabled="true"
-  apm_libraries=("datadog-apm-inject")
 fi
 
-docker_injection_enabled=
-if [[ "$DD_APM_DOCKER_INJECTION_ENABLED" == "true" || ( -z "$DD_APM_DOCKER_INJECTION_ENABLED" && "$DD_APM_INSTRUMENTATION_ENABLED" == "true") ]] ; then
+if [ "$DD_APM_INSTRUMENTATION_ENABLED" = "docker" ] ; then
   docker_injection_enabled="true"
-  apm_libraries=("datadog-apm-inject")
+  no_agent=true
+fi
+
+if [ -n "${host_injection_enabled}${docker_injection_enabled}" ] ; then
+    apm_libraries=("datadog-apm-inject")
+fi
+
+if [ -n "${host_injection_enabled}" ] && [ -n "${no_agent}" ] ; then
+    ERROR_MESSAGE="Host injection is not supported with the DD_NO_AGENT_INSTALL flag. Disable host injection with DD_APM_HOST_INJECTION_ENABLED=false or unset DD_NO_AGENT_INSTALL"
```

Comment:
I didn't notice this on the first review run, but I think the message is incorrect. I think it should say `Disable host injection with DD_APM_INSTRUMENTATION_ENABLED=docker ...` instead of `DD_APM_HOST_INJECTION_ENABLED=false`, shouldn't it?

Reply:
Text fixed.